rmw_zenoh_cpp/src/detail/rmw_client_data.cpp

```diff
@@ -458,23 +394,64 @@ rmw_ret_t ClientData::send_request(
   // Latest consolidation guarantees unicity of replies for the same key expression,
   // which optimizes bandwidth. The default is "None", which imples replies may come in any order
   // and any number.
-  opts.consolidation = z_query_consolidation_latest();
-  z_owned_bytes_t payload;
-  z_bytes_copy_from_buf(
-    &payload, reinterpret_cast<const uint8_t *>(request_bytes), data_length);
-  opts.payload = z_move(payload);
+  opts.consolidation = zenoh::ConsolidationMode::Z_CONSOLIDATION_MODE_NONE;
+
+  std::vector<uint8_t> raw_bytes(
+    reinterpret_cast<const uint8_t *>(request_bytes),
+    reinterpret_cast<const uint8_t *>(request_bytes) + data_length);
+  opts.payload = zenoh::Bytes(raw_bytes);
```

Comment:
We should use `zenoh::Bytes(std::move(raw_bytes))` to avoid a copy. See https://github.com/eclipse-zenoh/zenoh-cpp/blob/94edfd9845b63bab4761be328bd64db95daab212/include/zenoh/api/bytes.hxx?plain=1#L65 for details.